harness/determined/common/api/authentication.py

```diff
@@ -138,38 +138,18 @@ def get_session_token(self, must: bool = True) -> str:
         return self.session.token
 
 
-class AuthContext:
-    def __init__(self, token: str, user_id: str):
-        self.token = token
-        self.user_id = user_id
-
-    def get_user_id(self) -> str:
-        return self.user_id
-
-    def get_token(self) -> str:
-        return self.token
-
-
 def do_login(
     master_address: str,
     username: str,
     password: str,
     cert: Optional[certs.Cert] = None,
-) -> AuthContext:
-    r = api.post(
-        master_address,
-        "login",
-        json={"username": username, "password": password},
-        authenticated=False,
-        cert=cert,
+) -> bindings.v1LoginResponse:
+    certs.cli_cert = certs.default_load(master_address)
+    cli_auth = Authentication(master_address, try_reauth=True)
+    sess = det.common.experimental.session.Session(
+        master_address, "determined", cli_auth, certs.cli_cert
```

Comment:
we definitely can't hardcode the determined user. can we not make the request without a session or something initially? (login would be the only request this makes sense for, mostly)

Reply:
it looks like all the requests require a `session` for `session._do_request`. is that something i can just change in `bindings.py`, to use `api.get` instead? not sure if that's something that's okay for me to modify.

Reply:
It appears all the arguments to a session are optional, so I think you could just construct a session like
```
sess = det.common.experimental.session.Session(master_address, None, None, certs.cli_cert)
```

Reply:
maybe even `Session(master_address, None, None, None)`?
something similar happened recently in https://github.com/determined-ai/determined/pull/4096/files

but I am starting to get worried about proliferation of this entire pattern because it *must* be easier to make API calls without worrying about this.